src/main/java/com/google/sps/data/FirestoreDB.java

```diff
@@ -102,26 +102,34 @@ public Long voteRecipe(String Id, int voteDiff) {
         getHidden ? DBUtils.tags() : DBUtils.tags().whereEqualTo(Tag.HIDDEN_KEY, false);
     return DBUtils.blockOnFuture(tagsQuery.get()).toObjects(Tag.class);
   }
+
+  public List<Tag> getTagsMatchingIds(List<String> Ids) {
+    Query tagsQuery = DBUtils.tags().whereIn(Tag.ID_KEY, Ids);
+    return DBUtils.blockOnFuture(tagsQuery.get()).toObjects(Tag.class);
+  }
   
   public boolean isDocument(String docId, String collection) {
     DocumentReference docRef = DBUtils.database.collection(collection).document(docId);
     DocumentSnapshot document = DBUtils.blockOnFuture(docRef.get());
     return document.exists();
   }
 
+  /** Returns a User object from userId. */
```

Comment:
Why did you add these comments?

Reply:
I think they are from when I merged master into my branch. Ill delete them.